packages/x-date-pickers/src/DateField/DateField.types.ts

```diff
@@ -41,19 +40,6 @@ export interface UseDateFieldProps<
     BaseDateValidationProps<TDate>,
     ExportedUseClearableFieldProps {}
 
-/**
- * Props the field can receive when used inside a date picker.
- * (`DatePicker`, `DesktopDatePicker` or `MobileDatePicker` component).
- */
-export type DateFieldInPickerProps<
```

Comment:
I think it make a lot more sense to expose those at the picker level rather than at the field level.
That way, people creating fields with another DS won't have to import stuff from the component that they are not even using.

And for range picker it will allow me to create those interfaces in a follow up without settling on the multi input vs single input debate.

It is strictly speaking a BC, but those interfaces were introduces in #14486 and are never used in the doc, I think we can just remove them in a minor.